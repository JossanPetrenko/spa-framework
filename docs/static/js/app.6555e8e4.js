(window.webpackJsonp=window.webpackJsonp||[]).push([[1],{"./.docz/app/db.json":function(e){e.exports={config:{title:"Nimbus",description:"Nimbus - SPA Framework",themeConfig:{},menu:[],ordering:"descending",repository:null,native:!1,plugins:[{},{}],indexHtml:".docz/index_template.html",base:"/spa-framework/"},entries:{"packages/@nimbus/styles/readme.mdx":{name:"nimbus/styles",menu:"Api",id:"6293ed660430b98f5c0ad37f63823705",filepath:"packages/@nimbus/styles/readme.mdx",link:null,slug:"packages-nimbus-styles-readme",route:"/packages-nimbus-styles-readme",order:0,headings:[{depth:2,slug:"nimbusstyles",value:"@nimbus /styles"}]},"packages/@nimbus/button/docs/button.mdx":{name:"nimbus/button",menu:"Components",id:"94c0711901e304469ede6258c2b3c0f8",filepath:"packages/@nimbus/button/docs/button.mdx",link:null,slug:"packages-nimbus-button-docs-button",route:"/packages-nimbus-button-docs-button",order:0,headings:[{depth:1,slug:"button",value:"Button"},{depth:2,slug:"padr\xe3o",value:"Padr\xe3o"},{depth:3,slug:"exemplos",value:"Exemplos"},{depth:2,slug:"enfatizado",value:"Enfatizado"},{depth:3,slug:"exemplos-1",value:"Exemplos"},{depth:2,slug:"inten\xe7\xf5es",value:"Inten\xe7\xf5es"},{depth:3,slug:"exemplos-2",value:"Exemplos"},{depth:2,slug:"varia\xe7\xf5es",value:"Varia\xe7\xf5es"},{depth:3,slug:"exemplos---outlined",value:"Exemplos - Outlined"},{depth:3,slug:"exemplos---flat",value:"Exemplos - Flat"},{depth:2,slug:"\xcdcone",value:"\xcdcone"},{depth:3,slug:"exemplos-font-awesome-icons--material-desing-icons",value:"Exemplos (Font Awesome Icons / Material Desing Icons)"}]},"packages/@nimbus/entry/docs/entry.mdx":{name:"nimbus/entry",menu:"Components",id:"8c029e5f77ac97be9736b7b1e39ccc8b",filepath:"packages/@nimbus/entry/docs/entry.mdx",link:null,slug:"packages-nimbus-entry-docs-entry",route:"/packages-nimbus-entry-docs-entry",order:0,headings:[{depth:1,slug:"entry",value:"Entry"},{depth:2,slug:"text-entry",value:"Text Entry"},{depth:3,slug:"exemplo-simples",value:"Exemplo Simples"},{depth:3,slug:"exemplos-label",value:"Exemplos Label"},{depth:3,slug:"exemplos-disabled",value:"Exemplos Disabled"},{depth:3,slug:"exemplos-width",value:"Exemplos Width"},{depth:3,slug:"exemplos-icon",value:"Exemplos Icon"}]},"packages/@nimbus/entry/docs/label.mdx":{name:"nimbus/label",menu:"Components",id:"27d971dbc0734c71e55bad54fb466f07",filepath:"packages/@nimbus/entry/docs/label.mdx",link:null,slug:"packages-nimbus-entry-docs-label",route:"/packages-nimbus-entry-docs-label",order:0,headings:[{depth:1,slug:"entry-label",value:"Entry Label"},{depth:2,slug:"label",value:"Label"},{depth:3,slug:"exemplos",value:"Exemplos"}]},"packages/@nimbus/i18n/docs/i18n.mdx":{name:"nimbus/i18n",menu:"Api",id:"b659672b6ce7352e96793a2eee0efabf",filepath:"packages/@nimbus/i18n/docs/i18n.mdx",link:null,slug:"packages-nimbus-i18n-docs-i18n",route:"/packages-nimbus-i18n-docs-i18n",order:0,headings:[{depth:1,slug:"internationalization",value:"Internationalization"},{depth:2,slug:"uso-b\xe1sico",value:"Uso B\xe1sico"},{depth:2,slug:"v\xe1riaveis",value:"V\xe1riaveis"},{depth:2,slug:"parametros",value:"Parametros"},{depth:2,slug:"contagem-inteligente",value:"Contagem Inteligente"},{depth:2,slug:"mudan\xe7a-de-linguagem",value:"Mudan\xe7a de linguagem"},{depth:2,slug:"exemplo-no-react",value:"Exemplo no React"}]},"packages/@nimbus/layout/docs/sidemenu.mdx":{name:"nimbus/layout",menu:"Components",id:"ffb421ca5e14eeadc70208b9bafbc3bb",filepath:"packages/@nimbus/layout/docs/sidemenu.mdx",link:null,slug:"packages-nimbus-layout-docs-sidemenu",route:"/packages-nimbus-layout-docs-sidemenu",order:0,headings:[{depth:1,slug:"layout",value:"Layout"}]}}}},"./.docz/app/imports.js":function(e,n,s){"use strict";s.d(n,"a",function(){return a});var a={"packages/@nimbus/styles/readme.mdx":function(){return s.e(2).then(s.bind(null,"./packages/@nimbus/styles/readme.mdx"))},"packages/@nimbus/button/docs/button.mdx":function(){return Promise.all([s.e(0),s.e(3)]).then(s.bind(null,"./packages/@nimbus/button/docs/button.mdx"))},"packages/@nimbus/entry/docs/entry.mdx":function(){return Promise.all([s.e(0),s.e(4)]).then(s.bind(null,"./packages/@nimbus/entry/docs/entry.mdx"))},"packages/@nimbus/entry/docs/label.mdx":function(){return Promise.all([s.e(0),s.e(5)]).then(s.bind(null,"./packages/@nimbus/entry/docs/label.mdx"))},"packages/@nimbus/i18n/docs/i18n.mdx":function(){return s.e(6).then(s.bind(null,"./packages/@nimbus/i18n/docs/i18n.mdx"))},"packages/@nimbus/layout/docs/sidemenu.mdx":function(){return Promise.all([s.e(0),s.e(7)]).then(s.bind(null,"./packages/@nimbus/layout/docs/sidemenu.mdx"))}}},"./.docz/app/index.jsx":function(e,n,s){"use strict";s.r(n);var a=s("./node_modules/react/index.js"),t=s.n(a),u=s("./node_modules/react-dom/index.js"),l=s.n(u),o=s("./.docz/app/root.jsx"),d=[],i=[],m=function(){return i.forEach(function(e){return e&&e()})},c=document.querySelector("#root");!function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:o.a;d.forEach(function(e){return e&&e()}),l.a.render(t.a.createElement(e,null),c,m)}(o.a)},"./.docz/app/root.jsx":function(e,n,s){"use strict";(function(e){var a=s("./node_modules/react/index.js"),t=s.n(a),u=s("./node_modules/react-hot-loader/index.js"),l=s("./node_modules/docz-theme-default/dist/index.js"),o=s.n(l),d=s("./.docz/app/imports.js"),i=s("./.docz/app/db.json"),m=function(){return t.a.createElement(o.a,{db:i,imports:d.a,hashRouter:!1})};n.a=Object(u.hot)(e)(m),m.__docgenInfo={description:"",methods:[],displayName:"Root"}}).call(this,s("./node_modules/webpack/buildin/harmony-module.js")(e))},0:function(e,n,s){s("./node_modules/@babel/polyfill/lib/index.js"),e.exports=s("./.docz/app/index.jsx")},1:function(e,n){},2:function(e,n){},3:function(e,n){},4:function(e,n){}},[[0,8,0]],[0,3,4,5,7,6,2]]);