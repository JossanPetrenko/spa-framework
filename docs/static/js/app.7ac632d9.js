(window.webpackJsonp=window.webpackJsonp||[]).push([[1],{"./.docz/app/db.json":function(e){e.exports={config:{title:"Nimbus",description:"Nimbus - SPA Framework",themeConfig:{},menu:[],ordering:"descending",repository:null,native:!1,plugins:[{},{}],indexHtml:".docz/index_template.html"},entries:{"packages/@nimbus/button/button.mdx":{name:"nimbus/button",menu:"Components",id:"6f2239e3c6c70200bb02732d77430039",filepath:"packages/@nimbus/button/button.mdx",link:null,slug:"packages-nimbus-button-button",route:"/packages-nimbus-button-button",order:0,headings:[{depth:1,slug:"button",value:"Button"},{depth:2,slug:"padr\xe3o",value:"Padr\xe3o"},{depth:3,slug:"exemplos",value:"Exemplos"},{depth:2,slug:"enfatizado",value:"Enfatizado"},{depth:3,slug:"exemplos-1",value:"Exemplos"},{depth:2,slug:"inten\xe7\xf5es",value:"Inten\xe7\xf5es"},{depth:3,slug:"exemplos-2",value:"Exemplos"},{depth:2,slug:"varia\xe7\xf5es",value:"Varia\xe7\xf5es"},{depth:3,slug:"exemplos---outlined",value:"Exemplos - Outlined"},{depth:3,slug:"exemplos---flat",value:"Exemplos - Flat"},{depth:2,slug:"\xcdcone",value:"\xcdcone"},{depth:3,slug:"exemplos-font-awesome-icons--material-desing-icons",value:"Exemplos (Font Awesome Icons / Material Desing Icons)"}]},"packages/@nimbus/entry/entry.mdx":{name:"nimbus/entry",menu:"Components",id:"876a39f3bf5d2b6988e27feef58209df",filepath:"packages/@nimbus/entry/entry.mdx",link:null,slug:"packages-nimbus-entry-entry",route:"/packages-nimbus-entry-entry",order:0,headings:[{depth:1,slug:"entry",value:"Entry"},{depth:2,slug:"text-entry",value:"Text Entry"},{depth:3,slug:"exemplo-simples",value:"Exemplo Simples"},{depth:3,slug:"exemplos-label",value:"Exemplos Label"},{depth:3,slug:"exemplos-disabled",value:"Exemplos Disabled"},{depth:3,slug:"exemplos-width",value:"Exemplos Width"},{depth:3,slug:"exemplos-icon",value:"Exemplos Icon"}]},"packages/@nimbus/entry/label.mdx":{name:"nimbus/label",menu:"Components",id:"35b18823f39ebeb545340fb9da304f91",filepath:"packages/@nimbus/entry/label.mdx",link:null,slug:"packages-nimbus-entry-label",route:"/packages-nimbus-entry-label",order:0,headings:[{depth:1,slug:"entry-label",value:"Entry Label"},{depth:2,slug:"label",value:"Label"},{depth:3,slug:"exemplos",value:"Exemplos"}]},"packages/@nimbus/i18n/i18n.mdx":{name:"nimbus/i18n",menu:"Api",id:"57c1624df491bb28d8c6b12b20cd6bd3",filepath:"packages/@nimbus/i18n/i18n.mdx",link:null,slug:"packages-nimbus-i18n-i18n",route:"/packages-nimbus-i18n-i18n",order:0,headings:[{depth:1,slug:"internationalization",value:"Internationalization"},{depth:2,slug:"uso-b\xe1sico",value:"Uso B\xe1sico"},{depth:2,slug:"v\xe1riaveis",value:"V\xe1riaveis"},{depth:2,slug:"parametros",value:"Parametros"},{depth:2,slug:"contagem-inteligente",value:"Contagem Inteligente"},{depth:2,slug:"mudan\xe7a-de-linguagem",value:"Mudan\xe7a de linguagem"},{depth:2,slug:"exemplo-no-react",value:"Exemplo no React"}]},"packages/@nimbus/styles/readme.mdx":{name:"nimbus/styles",menu:"Api",id:"6293ed660430b98f5c0ad37f63823705",filepath:"packages/@nimbus/styles/readme.mdx",link:null,slug:"packages-nimbus-styles-readme",route:"/packages-nimbus-styles-readme",order:0,headings:[{depth:2,slug:"nimbusstyles",value:"@nimbus /styles"}]}}}},"./.docz/app/imports.js":function(e,n,s){"use strict";s.d(n,"a",function(){return t});var t={"packages/@nimbus/button/button.mdx":function(){return Promise.all([s.e(0),s.e(2)]).then(s.bind(null,"./packages/@nimbus/button/button.mdx"))},"packages/@nimbus/entry/entry.mdx":function(){return Promise.all([s.e(0),s.e(3)]).then(s.bind(null,"./packages/@nimbus/entry/entry.mdx"))},"packages/@nimbus/entry/label.mdx":function(){return Promise.all([s.e(0),s.e(4)]).then(s.bind(null,"./packages/@nimbus/entry/label.mdx"))},"packages/@nimbus/i18n/i18n.mdx":function(){return s.e(5).then(s.bind(null,"./packages/@nimbus/i18n/i18n.mdx"))},"packages/@nimbus/styles/readme.mdx":function(){return s.e(6).then(s.bind(null,"./packages/@nimbus/styles/readme.mdx"))}}},"./.docz/app/index.jsx":function(e,n,s){"use strict";s.r(n);var t=s("./node_modules/react/index.js"),a=s.n(t),l=s("./node_modules/react-dom/index.js"),u=s.n(l),o=s("./.docz/app/root.jsx"),i=[],d=[],m=function(){return d.forEach(function(e){return e&&e()})},p=document.querySelector("#root");!function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:o.a;i.forEach(function(e){return e&&e()}),u.a.render(a.a.createElement(e,null),p,m)}(o.a)},"./.docz/app/root.jsx":function(e,n,s){"use strict";(function(e){var t=s("./node_modules/react/index.js"),a=s.n(t),l=s("./node_modules/react-hot-loader/index.js"),u=s("./node_modules/docz-theme-default/dist/index.js"),o=s.n(u),i=s("./.docz/app/imports.js"),d=s("./.docz/app/db.json"),m=function(){return a.a.createElement(o.a,{db:d,imports:i.a,hashRouter:!1})};n.a=Object(l.hot)(e)(m),m.__docgenInfo={description:"",methods:[],displayName:"Root"}}).call(this,s("./node_modules/docz/node_modules/webpack/buildin/harmony-module.js")(e))},0:function(e,n,s){s("./node_modules/@babel/polyfill/lib/index.js"),e.exports=s("./.docz/app/index.jsx")},1:function(e,n){},2:function(e,n){},3:function(e,n){},4:function(e,n){}},[[0,7,0]],[0,2,3,4,5,6]]);