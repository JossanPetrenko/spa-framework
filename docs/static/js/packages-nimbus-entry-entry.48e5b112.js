(window.webpackJsonp=window.webpackJsonp||[]).push([[3],{"./packages/@nimbus/entry/entry.mdx":function(e,o,l){"use strict";l.r(o);var a=l("./node_modules/react/index.js"),n=l.n(a),c=l("./node_modules/@mdx-js/tag/dist/index.js"),t=(l("./packages/@nimbus/styles/scss/theme.scss"),l("./packages/@nimbus/strings/index.js")),m=l("./node_modules/react-grid-system/build/index.js"),d=l("./node_modules/docz/dist/index.m.js"),r=l("./packages/@nimbus/entry/index.js");function p(e,o){if(null==e)return{};var l,a,n=function(e,o){if(null==e)return{};var l,a,n={},c=Object.keys(e);for(a=0;a<c.length;a++)l=c[a],o.indexOf(l)>=0||(n[l]=e[l]);return n}(e,o);if(Object.getOwnPropertySymbols){var c=Object.getOwnPropertySymbols(e);for(a=0;a<c.length;a++)l=c[a],o.indexOf(l)>=0||Object.prototype.propertyIsEnumerable.call(e,l)&&(n[l]=e[l])}return n}o.default=function(e){var o=e.components,l=p(e,["components"]);return n.a.createElement(c.MDXTag,{name:"wrapper",components:o},n.a.createElement(c.MDXTag,{name:"h1",components:o,props:{id:"entry"}},n.a.createElement(c.MDXTag,{name:"strong",components:o,parentName:"h1"},"Entry")),n.a.createElement(d.PropsTable,{of:r.b}),n.a.createElement(c.MDXTag,{name:"h2",components:o,props:{id:"text-entry"}},"Text Entry"),n.a.createElement(c.MDXTag,{name:"h3",components:o,props:{id:"exemplo-simples"}},"Exemplo Simples"),n.a.createElement(d.Playground,{__position:1,__codesandbox:"undefined",__code:"<Text label={t('common.label')} placeholder={t('common.placeholder')} />",__scope:{props:l,t:t.a,Row:m.Row,Col:m.Col,Text:r.b}},n.a.createElement(r.b,{label:Object(t.a)("common.label"),placeholder:Object(t.a)("common.placeholder")})),n.a.createElement(c.MDXTag,{name:"h3",components:o,props:{id:"exemplos-label"}},"Exemplos Label"),n.a.createElement(d.Playground,{__position:2,__codesandbox:"undefined",__code:"<Row>\n  <Col md={4}>\n    <Text label={t('common.label')} placeholder={t('common.placeholder')} />\n  </Col>\n  <Col md={4}>\n    <Text label placeholder={t('common.placeholder')} />\n  </Col>\n\n  <Col md={4}>\n    <Text placeholder={t('common.placeholder')} />\n  </Col>\n</Row>",__scope:{props:l,t:t.a,Row:m.Row,Col:m.Col,Text:r.b}},n.a.createElement(m.Row,null,n.a.createElement(m.Col,{md:4},n.a.createElement(r.b,{label:Object(t.a)("common.label"),placeholder:Object(t.a)("common.placeholder")})),n.a.createElement(m.Col,{md:4},n.a.createElement(r.b,{label:!0,placeholder:Object(t.a)("common.placeholder")})),n.a.createElement(m.Col,{md:4},n.a.createElement(r.b,{placeholder:Object(t.a)("common.placeholder")})))),n.a.createElement(c.MDXTag,{name:"h3",components:o,props:{id:"exemplos-disabled"}},"Exemplos Disabled"),n.a.createElement(d.Playground,{__position:3,__codesandbox:"undefined",__code:"<Text\n  label={t('common.label')}\n  placeholder={t('common.placeholder')}\n  disabled\n/>",__scope:{props:l,t:t.a,Row:m.Row,Col:m.Col,Text:r.b}},n.a.createElement(r.b,{label:Object(t.a)("common.label"),placeholder:Object(t.a)("common.placeholder"),disabled:!0})),n.a.createElement(c.MDXTag,{name:"h3",components:o,props:{id:"exemplos-width"}},"Exemplos Width"),n.a.createElement(d.Playground,{__position:4,__codesandbox:"undefined",__code:"<Text\n  label={t('common.label')}\n  placeholder={t('common.placeholder')}\n  width=\"50%\"\n/>\n<Text\n  label={t('common.label')}\n  placeholder={t('common.placeholder')}\n  width=\"392px\"\n/>\n<Text\n  label={t('common.label')}\n  placeholder={t('common.placeholder')}\n  width=\"49rem\"\n/>",__scope:{props:l,t:t.a,Row:m.Row,Col:m.Col,Text:r.b}},n.a.createElement(r.b,{label:Object(t.a)("common.label"),placeholder:Object(t.a)("common.placeholder"),width:"50%"}),n.a.createElement(r.b,{label:Object(t.a)("common.label"),placeholder:Object(t.a)("common.placeholder"),width:"392px"}),n.a.createElement(r.b,{label:Object(t.a)("common.label"),placeholder:Object(t.a)("common.placeholder"),width:"49rem"})),n.a.createElement(c.MDXTag,{name:"h3",components:o,props:{id:"exemplos-icon"}},"Exemplos Icon"),n.a.createElement(d.Playground,{__position:5,__codesandbox:"undefined",__code:"<Text\n  label={t('common.label')}\n  placeholder={t('common.placeholder')}\n  iconLeft=\"fas fa-cloud\"\n/>\n<Text\n  label={t('common.label')}\n  placeholder={t('common.placeholder')}\n  iconRight=\"fas fa-cloud\"\n/>\n<Text\n  label={t('common.label')}\n  placeholder={t('common.placeholder')}\n  iconRight=\"fas fa-cloud\"\n  iconLeft=\"fas fa-cloud\"\n/>",__scope:{props:l,t:t.a,Row:m.Row,Col:m.Col,Text:r.b}},n.a.createElement(r.b,{label:Object(t.a)("common.label"),placeholder:Object(t.a)("common.placeholder"),iconLeft:"fas fa-cloud"}),n.a.createElement(r.b,{label:Object(t.a)("common.label"),placeholder:Object(t.a)("common.placeholder"),iconRight:"fas fa-cloud"}),n.a.createElement(r.b,{label:Object(t.a)("common.label"),placeholder:Object(t.a)("common.placeholder"),iconRight:"fas fa-cloud",iconLeft:"fas fa-cloud"})))}}}]);