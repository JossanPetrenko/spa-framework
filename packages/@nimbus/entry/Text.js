import React from 'react';
import PropTypes from 'prop-types';
import {css} from '@nimbus/core';
import Label from './Label';
import uuid from 'uuid/v4';

export default class Text extends React.PureComponent {
  constructor (props) {
    super (props);
    this.state = {id: props.id || `Text-${uuid ()}`};
  }

  renderInput () {
    const {
      children,
      placeholder,
      label,
      width,

      ...othersProps
    } = this.props;

    const classList = css ('n-entry-text');

    return (
      <input
        className={classList}
        placeholder={placeholder}
        id={this.state.id}
        {...othersProps}
      />
    );
  }
  render () {
    const classRoot = css ('n-entry-root');
    const {label, width} = this.props;

    if ('label' in this.props) {
      return (
        <div style={{width, boxSizing: 'border-box'}} className={classRoot}>
          <Label htmlFor={this.state.id}>
            {label}
          </Label>
          {this.renderInput ()}
        </div>
      );
    } else {
      return (
        <div style={{width, boxSizing: 'border-box'}} className={classRoot}>
          {this.renderInput ()}
        </div>
      );
    }
  }
}

Text.propTypes = {
  /** RÃ³tulo do input */
  label: PropTypes.string,

  /** Placeholder do input */
  placeholder: PropTypes.string,

  /** Input desabilitado */
  disabled: PropTypes.bool,

  /** Tamanho horizontal do input */
  width: PropTypes.string,
};

Text.defaultProps = {
  width: '100%',
};
