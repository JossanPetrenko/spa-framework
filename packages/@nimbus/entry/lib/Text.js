function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

import React from 'react';
import PropTypes from 'prop-types';
import { css } from '@nimbus/core';
import Label from './Label';
import uuid from 'uuid/v4';

var Text =
/*#__PURE__*/
function (_React$PureComponent) {
  _inherits(Text, _React$PureComponent);

  function Text(props) {
    var _this;

    _classCallCheck(this, Text);

    _this = _possibleConstructorReturn(this, _getPrototypeOf(Text).call(this, props));
    _this.state = {
      id: props.id || "Text-".concat(uuid())
    };
    return _this;
  }

  _createClass(Text, [{
    key: "handleIcon",
    value: function handleIcon(icon) {
      var orientation = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 'right';
      if (!icon) return;
      return React.createElement("span", {
        className: "n-entry-icon ".concat(icon, " ").concat(orientation)
      });
    }
  }, {
    key: "renderInput",
    value: function renderInput() {
      var _this$props = this.props,
          children = _this$props.children,
          placeholder = _this$props.placeholder,
          label = _this$props.label,
          width = _this$props.width,
          iconRight = _this$props.iconRight,
          iconLeft = _this$props.iconLeft,
          othersProps = _objectWithoutProperties(_this$props, ["children", "placeholder", "label", "width", "iconRight", "iconLeft"]);

      var classInput = css('n-entry');
      return React.createElement("span", null, React.createElement("input", _extends({
        className: classInput,
        placeholder: placeholder,
        id: this.state.id
      }, othersProps)), this.handleIcon(iconLeft, 'left'), this.handleIcon(iconRight, 'right'));
    }
  }, {
    key: "render",
    value: function render() {
      var classRoot = css('n-entry-root');
      var classContainer = css('n-entry-container', {
        'n-input-icon-left': this.props.iconLeft,
        'n-input-icon-right': this.props.iconRight
      });
      var _this$props2 = this.props,
          label = _this$props2.label,
          width = _this$props2.width;

      if ('label' in this.props) {
        return React.createElement("div", {
          className: classRoot,
          style: {
            width: width
          }
        }, React.createElement(Label, {
          htmlFor: this.state.id
        }, label), React.createElement("div", {
          className: classContainer
        }, this.renderInput()));
      } else {
        return React.createElement("div", {
          style: {
            width: width,
            boxSizing: 'border-box'
          },
          className: classRoot
        }, this.renderInput());
      }
    }
  }]);

  return Text;
}(React.PureComponent);

export { Text as default };
Text.propTypes = {
  /** Rótulo do input. */
  label: PropTypes.PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),

  /** Placeholder do input. */
  placeholder: PropTypes.string,

  /** Input desabilitado. */
  disabled: PropTypes.bool,

  /** Tamanho horizontal do input. */
  width: PropTypes.string,

  /** Ícone no entry a esquerda. */
  iconLeft: PropTypes.string,

  /** Ícone no entry a direita. */
  iconRight: PropTypes.string
};
Text.defaultProps = {
  width: '100%'
};