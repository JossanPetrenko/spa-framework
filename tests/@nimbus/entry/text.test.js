import React from 'react';
import {expect} from 'chai';
import {t} from '../../../packages/@nimbus/strings/index.js';

import Text from '../../../packages/@nimbus/entry/Text';

it ('deve renderizar o input text com label', () => {
  const wrapper = shallow (
    <Text id='Nome' text={t ('common.titulo')} placeholder={t ('common.placeholder')} />
  );
  expect (wrapper).to.matchSnapshot ();
});
