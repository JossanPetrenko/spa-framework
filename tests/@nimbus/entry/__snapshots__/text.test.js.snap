// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deve renderizar o input text com label 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Text
    id="Nome"
    placeholder="Placeholder"
    text="Título"
    width="100%"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        className="n-entry-text"
        id="Nome"
        placeholder="Placeholder"
        text="Título"
      />,
      "className": "n-entry-root",
      "style": Object {
        "boxSizing": "border-box",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "n-entry-text",
        "id": "Nome",
        "placeholder": "Placeholder",
        "text": "Título",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          className="n-entry-text"
          id="Nome"
          placeholder="Placeholder"
          text="Título"
        />,
        "className": "n-entry-root",
        "style": Object {
          "boxSizing": "border-box",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "n-entry-text",
          "id": "Nome",
          "placeholder": "Placeholder",
          "text": "Título",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
